tests/serverpod_test_server/test_e2e/json_protocol_test.dart

```diff
@@ -6,6 +7,29 @@ import 'package:test/expect.dart';
 import 'package:test/scaffolding.dart';
 import 'package:web_socket_channel/web_socket_channel.dart';
 
+Future<dynamic> _getWebsocketMessage(
+  WebSocketChannel websocket,
+) async {
+  try {
+    return await websocket.stream
+        .timeout(
+          Duration(seconds: 5),
+          onTimeout: (sink) => throw TimeoutException(
+            'Message was not received within the timeout period.',
+          ),
+        )
+        .firstWhere(
+          (event) =>
+              event is String && event.contains('serverOnlyScopedFieldModel'),
+          orElse: () => throw TimeoutException(
+            'Message was not received within the timeout period.',
+          ),
```

Comment:
Question: Is this or else still needed? Won't the timeout throw regardless of this? might be that this statement is only executed if the stream is closed or done before the `where` condition is fullfilled.

Reply:
You're correct. I'll return null here because if no item is found, the operation will eventually time out. Essentially, it will wait indefinitely for the condition to be met, leading to a timeout.